docs/modules/ROOT/pages/api/security.adoc

```diff
@@ -0,0 +1,200 @@
+* xref:#PausableComponent-_pause[`++_pause(self)++`]
+* xref:#PausableComponent-_unpause[`++_unpause(self)++`]
+--
+
+[.contract-index]
+.Events
+--
+* xref:#PausableComponent-Paused[`++Paused(account)++`]
+* xref:#PausableComponent-Unpaused[`++Unpaused(account)++`]
+--
+
+[#PausableComponent-Embeddable-Functions]
+==== Embeddable Functions
+
+[.contract-item]
+[[PausableComponent-is_paused]]
+==== `[.contract-item-name]#++is_paused++#++(self: @ComponentState)++ → bool` [.item-kind]#external#
+
+Returns whether the contract is currently on pause.
+
+[#PausableComponent-Internal-Functions]
+==== Internal Functions
+
+[.contract-item]
+[[PausableComponent-assert_not_paused]]
+==== `[.contract-item-name]#++assert_not_paused++#++(self: @ComponentState)++` [.item-kind]#internal#
+
+Makes a function only callable when the contract is not paused.
+
+[.contract-item]
+[[PausableComponent-assert_paused]]
+==== `[.contract-item-name]#++assert_paused++#++(self: @ComponentState)++` [.item-kind]#internal#
+
+Makes a function only callable when the contract is paused.
+
+[.contract-item]
+[[PausableComponent-_pause]]
+==== `[.contract-item-name]#++_pause++#++(ref self: ComponentState)++` [.item-kind]#internal#
+
+Pauses the contract.
+
+Requirements:
+
+- the contract must not be paused.
+
+Emits a {Paused} event.
+
+[.contract-item]
+[[PausableComponent-_unpause]]
+==== `[.contract-item-name]#++_unpause++#++(ref self: ComponentState)++` [.item-kind]#internal#
+
+Unpauses the contract.
+
+Requirements:
+
+- the contract must be paused.
+
+Emits an {Unpaused} event.
+
+
+[#PausableComponent-Events]
+==== Events
+
+[.contract-item]
+[[PausableComponent-Paused]]
+==== `[.contract-item-name]#++Paused++#++(account: ContractAddress)++` [.item-kind]#event#
+
+Emitted when the pause is triggered by `account`.
+
+[.contract-item]
+[[PausableComponent-Unpaused]]
+==== `[.contract-item-name]#++Unpaused++#++(account: ContractAddress)++` [.item-kind]#event#
+
+Emitted when the pause is lifted by `account`.
+
+== ReentrancyGuard
+
+[.contract]
+[[ReentrancyGuardComponent]]
+=== `++ReentrancyGuardComponent++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.7.0/src/security/reentrancyguard.cairo[{github-icon},role=heading-link]
+
+```javascript
+use openzeppelin::security::ReentrancyGuardComponent;
+```
+
+Component to help prevent nested (reentrant) calls.
+
+[.contract-index]
+.Internal Implementations
+--
+.InternalImpl
+
+* xref:#ReentrancyGuardComponent-start[`++start(self)++`]
+* xref:#ReentrancyGuardComponent-end[`++end(self)++`]
+--
+
+[#ReentrancyGuardComponent-Internal-Functions]
+==== Internal Functions
+
+[.contract-item]
+[[ReentrancyGuardComponent-start]]
+==== `[.contract-item-name]#++start++#++(ref self: ComponentState)++` [.item-kind]#internal#
+
+Prevents a contract's function from calling itself or another protected function, directly or indirectly.
+
+Requirements:
+
+- the contract must not be currently in a guarded state.
```

Comment:
```suggestion
Requires the `ReentrancyGuard` to not be currently enabled.
```
what do you think?

Reply:
The sentence read better, but as [mentioned before](https://github.com/OpenZeppelin/cairo-contracts/pull/794#discussion_r1389317656), I think the single line requirement could be a separate issue and PR.